test/extended/router/metrics.go

```diff
@@ -134,7 +136,7 @@ var _ = g.Describe("[Conformance][Area:Networking][Feature:Router]", func() {
 					}
 					// send a burst of traffic to the router
 					g.By("sending traffic to a weighted route")
-					err = expectRouteStatusCodeRepeatedExec(ns, execPodName, fmt.Sprintf("http://%s", serviceIP), "weighted.metrics.example.com", http.StatusOK, times)
+					err = expectRouteStatusCodeRepeatedExec(ns, execPodName, fmt.Sprintf("http://%s", ingressEndpoint), "weighted.metrics.example.com", http.StatusOK, times)
```

Comment:
Why use `ingressEndpoint` rather than `routerIP`?

Reply:
Depending on the platform, the ingress controller might be speaking PROXY. To portably exercise a a real route in the context of a black box test we need to be somewhat contextual (e.g. go through the external LB).

IMO all these router tests needs shuffled/relocated/refactored. Things that need to talk directly to a router probably belong in openshift/router e2e tests. Stuff that remains in the origin suite should be using route hostnames.

Reply:
On second thought, doesn't this mean the traffic is going to be balanced between the routers?

Reply:
Maybe this test is a good candidate for biting the bullet and moving over to openshift/router?

Reply:
> On second thought, doesn't this mean the traffic is going to be balanced between the routers?

Yeah.  However, the test keeps sending traffic to the service till the router from which it gathers metrics reports receiving traffic, so it should be all right:  https://github.com/openshift/origin/blob/f9d13286aae646b1e5ea9fe26695c1dbfcd7fcc6/test/extended/router/metrics.go#L125-L134

Reply:
Seems brittle, but as long as the test is now stable, is it worth going ahead like this to restore coverage right away?

Reply:
I think so.

Reply:
Okay.

/hold cancel

Reply:
The test failure in https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/23431/pull-ci-openshift-origin-master-e2e-aws/12031 looks like a failure waiting for traffic to be received by the expected router — given we don't have control over the balancing behavior, isn't this a new source of flakes?